test/MarketInvariants.t.sol

```diff
@@ -0,0 +1,87 @@
+/* solhint-disable contract-name-camelcase, func-name-mixedcase, reason-string */
```

Comment:
This is the file that actually defines the invariants (so the contracts defined in here are what would be called like forge test --mc MarketInvariantTest).